First commit of the new biojava-spark functions.

abradle opened the pull request.

abradle committed:
First commit of the new biojava-spark functions.

andreasprlic commented on src/main/java/org/biojava/examples/SimpleExample.java:
```diff
@@ -0,0 +1,39 @@
+		
+		Long startTime = System.currentTimeMillis();
+		// Get the RDD of the atom contacts
+		AtomDataRDD atomDataRDD = new StructureDataRDD("/Users/anthony/full")
```
Need to find a better way to configure this location

abradle replied:
Yes - that's a good point.

abradle replied:
I was thinking it would also be cool to provide a small (50mb) hadoop file as part of the JAR - so people can get going without  a download.

abradle committed:
Refactored the AtomSelectObject to use the Builder Pattern

abradle committed:
Added some more chain related and C-alpha related functions to Biojava

abradle committed:
Added an example of counting contacts

abradle committed:
Added another couple of examples and cleaned up the existing ones

abradle committed:
Clean up the C-alpha example.

abradle committed:
Added a resource of a mini-hadoop file
Updated the functions to work with this.

abradle committed:
Added ability to set the File path to read from the utils class.

abradle committed:
Added a download PDB example (not quite working yet).

abradle committed:
Added the MMTF dependency

abradle committed:
Added the encoder

abradle committed:
Added builder for uber jar

abradle committed:
Merge remote-tracking branch 'origin/master'

abradle committed:
Farily extensive refactoring of the code.
Pyspark entry point and handling.
New concept of a Segment and generation of fragments.

abradle committed:
Added  a new SegmentData class and some example reduce code.

abradle committed:
Fixed a compile error

abradle committed:
Another chain example and some updates to the Segment RDD

abradle committed:
Added a function to group by sequence

Untested...

abradle committed:
Addition of SegmentClustering and an example of how to use that.

abradle committed:
Fairly meaty refactoring of the code - and removal of the non biojava spark stuff.
